src/tr/merlinscans/src/eu/kanade/tachiyomi/extension/tr/merlinscans/MerlinScans.kt

```diff
@@ -1,12 +1,451 @@
+            title = element.select("div.series-title").text()
+            thumbnail_url = element.select("img").attr("src")
+        }
+    }
+
+    override fun popularMangaNextPageSelector() = "li:not(.disabled) a i.fa-chevron-right"
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        return GET(baseUrl, headers)
+    }
+
+    override fun latestUpdatesSelector() =
+        "div.latest-chapters-grid div.latest-series-horizontal-card"
+
+    override fun latestUpdatesFromElement(element: Element): SManga {
+        return SManga.create().apply {
+            val seriesLink = element.select("h3.series-title a").attr("href")
+            setUrlWithoutDomain(normalizeUrl(seriesLink))
+            title = element.select("h3.series-title a").text()
+            thumbnail_url = element.select("div.series-cover-thumbnail img").attr("src")
+        }
+    }
+
+    override fun latestUpdatesNextPageSelector(): String? = null
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        if (query.isNotEmpty()) {
+            val url = "$baseUrl/ajax/search-preview.php".toHttpUrl().newBuilder()
+                .addQueryParameter("q", query)
+                .build()
+            return GET(url, headers)
+        }
+
+        val url = "$baseUrl/all-series.php".toHttpUrl().newBuilder()
+            .addQueryParameter("page", page.toString())
+
+        var sortApplied = false
+
+        filters.forEach { filter ->
+            when (filter) {
+                is TypeFilter -> {
+                    if (filter.state != 0) {
+                        url.addQueryParameter("type", filter.toUriPart())
+                    }
+                }
+
+                is CategoryFilter -> {
+                    if (filter.state != 0) {
+                        url.addQueryParameter("category", filter.toUriPart())
+                    }
+                }
+
+                is StatusFilter -> {
+                    if (filter.state != 0) {
+                        url.addQueryParameter("status", filter.toUriPart())
+                    }
+                }
+
+                is SortFilter -> {
+                    if (filter.state != 0) {
+                        url.addQueryParameter("sort", filter.toUriPart())
+                        sortApplied = true
+                    }
+                }
+
+                else -> {}
+            }
+        }
+
+        if (!sortApplied) {
+            url.addQueryParameter("sort", "views")
+        }
+
+        return GET(url.build(), headers)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val requestUrl = response.request.url.toString()
+
+        if (requestUrl.contains("search-preview.php")) {
+            val body = response.body.string()
+            return parseSearchJson(body)
+        } else {
+            val document = response.asJsoup()
+            return parseSearchPage(document, response)
```

Comment:
just call super method, which calls the associated selector and parse methods
```suggestion
            return super.searchMangaParse(response)
```